flathunter/crawler/kleinanzeigen.py

```diff
@@ -73,27 +73,31 @@ def extract_data(self, soup):
             address = " ".join(address.split())
 
             rooms = ""
-            if len(tags) > 1:
-                rooms_match = re.search(r'\d+[.|,]*\d*', tags[1].text, flags=re.MULTILINE)
+            if len(tags) == 1:
+                rooms_match = re.search(r'\d[,|.]*(\d)*(\s)*Zi', tags[0].text, flags=re.MULTILINE)
                 if rooms_match is not None:
                     rooms = rooms_match.group()
+                    rooms = rooms.split(" ")[0]
 
-            try:
-                size = tags[0].text.strip()
-            except (IndexError, TypeError):
-                size = ""
+            size = ""
+            if len(tags) == 1:
+                size_match = re.search(r'\d+[,|.]*(\d)*', tags[0].text, flags=re.MULTILINE)
+                if size_match is not None:
+                    size = size_match.group()
+                    size = size.split(" ")[0]
 
             details = {
                 'id': int(expose.get("data-adid")),
                 'image': image,
                 'url': ("https://www.kleinanzeigen.de" + url),
                 'title': title_elem.text.strip(),
                 'price': price,
-                'size': size,
+                'size': size + "m²",
```

Comment:
I'm not so sure about putting the "m²" here. I would prefer the exposé just to have the raw data and add the unit in the message template. Is there a reason you want to have the unit here in the details?